source/extensions/common/wasm/context.cc

```diff
@@ -1305,6 +1305,32 @@ WasmResult Context::continueStream(StreamType stream_type) {
   return WasmResult::Ok;
 }
 
+WasmResult Context::closeStream(WasmStreamType stream_type) {
+  switch (stream_type) {
+  case WasmStreamType::Request:
+    if (decoder_callbacks_) {
+      decoder_callbacks_->resetStream();
+    }
+    return WasmResult::Ok;
+  case WasmStreamType::Response:
+    if (encoder_callbacks_) {
+      encoder_callbacks_->resetStream();
```

Comment:
Nit: You probably should reset both streams in both cases.

Reply:
That is a very 1:1 mindset. A more general model is a transformer multiplexer model which is not strictly 1:1.  When we merge with the other WASI proposal the model will have to become more general so best not to bake a restrictive model in now.

Reply:
This is/was about envoy-wasm implementation, where 1:1 relation is baked, and not WASI standard.